### PR TITLE
Update for the latest Binaryen API

### PR DESCRIPTION
**Pull request by MaxDesiatov**
A few things were removed upstream, so they are cleaned up here as well. The only opinionated change in this PR is that `Op` is renamed to `Instruction` for clarity, and is also more consistent with the WebAssembly spec. `RawRepresentable` is added on `Instruction` and `Type`, and the underlying value property is renamed accordingly.

Unfortunately, looks like Homebrew `.dylib` files detection is broken in SwiftPM, this only builds for me with `-Xlinker -lbinaryen -Xlinker -L/usr/local/lib` flags, which makes it not very useful as is. I probably will copy this code with appropriate attribution to my Binaryen repository fork, but I think this still should be submitted to your repository as a PR.

The test passes with `swift test -Xlinker -lbinaryen -Xlinker -L/usr/local/lib`, I can submite a separate PR with GitHub Actions CI if you're interested.

### CODE REVIEW

File: Tests/BinaryenTests/BinaryenTests.swift [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
+             (func $adder (param $0 i32) (param $1 i32) (result i32)
+              (i32.add
+               (local.get $0)
+               (local.get $1)
+              )
+             )
+            )
+            
+            """
+        )
```
Comment (turbolent) [inline Tests/BinaryenTests/BinaryenTests.swift:32]: 👍

**Review by turbolent — APPROVED** [review]
@MaxDesiatov Great work, thank you for bringing this up-to-date and making it more idiomatic! 👏 

Regarding the linking: I usually used `--xcconfig-overrides binaryen.xcconfig`, but agreed, it would be nice if this would just work without extra flags at some point!

**Comment by turbolent** [comment]
Sorry about the delay, I totally had missed this PR! I never got any PRs on my private projects before 😄  

Also, it would be great to have GitHub Actions set up for this! 💯

**Comment by MaxDesiatov** [comment]
Thanks for the review and merging!

**Pull request merged**